FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -643,10 +650,17 @@ - (void)setAPNSToken:(NSData *)APNSToken withUserInfo:(NSDictionary *)userInfo {
     return;
   }
   // Use this token type for when we have to automatically fetch tokens in the future
+#if TARGET_OS_SIMULATOR && TARGET_OS_IOS
+  // If APNS token is available on iOS Simulator, we must use the sandbox profile
+  // https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes
+  BOOL isSandboxApp = YES;
+#else
+  NSInteger type = [userInfo[kFIRMessagingAPNSTokenType] integerValue];
```

Comment:
would this always overwrite isSandboxApp? e.g. for simulator situation, would it be possible that userInfo[kFIRMessagingAPNSTokenType] doesn't exist?